src/main/java/com/google/sps/servlets/CheckPermissionsServlet.java

```diff
@@ -45,7 +45,7 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String projectId = request.getParameter("projID");
-    File file = new File(projectId+"-key.json");
+    File file = new File(projectId+".json");
```

Comment:
Make sure this is not vulnerable to a Path Traversal attack (https://owasp.org/www-community/attacks/Path_Traversal). For example what happens if request.getParameter("projID") == "/etc/passwd"?

More generally, never trust user input: https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html

Reply:
Good point - I'll have a look at improving that section tomorrow. Thank you!